### ISSUE TITLE
amd64 missing?

### ISSUE THREAD

**Issue by robklg** [rust-lang/docker-rust#147]
amd64 architecture appears to missing for some of the images in dockerhub, like 1.70.0-slim

**Comment by t-matsudate** [comment]
Also I've got this issue in GitHub Actions.

> Pull down action image 'rust:1.70.0'
>  /usr/bin/docker pull rust:1.70.0
>  no matching manifest for linux/amd64 in the manifest list entries
>  1.70.0: Pulling from library/rust
>  Warning: Docker pull failed with exit code 1, back off 9.944 seconds before retry.

**Comment by yaa110** [comment]
same issue [here](https://hub.docker.com/layers/library/rust/1.70.0-bullseye/images/sha256-0d14a397caaf5ae6ed907e56a1dbe274ca7d2958ea30943d167d339f3e16cff5?context=explore):

```
no matching manifest for linux/amd64 in the manifest list entries
```

**Comment by sfackler** [comment]
The images are built by Docker Hub infrastructure - we have no control over when that happens.

**Comment by tristan-morris** [comment]
I haven't looked at this before, but clicked a bunch of buttons on Docker hub... You can view the status of the official builds at https://doi-janky.infosiftr.net/ running on a Jenkins. It looks like the build jobs have completed for i386 and arm, whereas the amd64 builds started on 01 June and are still running (not for rust, that's done but it's 1.69). The amd64 pipeline is building all the things from 01 June (rabbit, debian, whatever..). 1.70 was also released on 01 June, so assume it just missed the cutoff. There's another build run scheduled, so assume 1.70 amd64 will get picked up then - doesn't appear to have errored. I think one of the dependent layers updated so Jenkins is sad and needs some more cpuz

Trigger job: https://doi-janky.infosiftr.net/job/multiarch/job/amd64/job/_trigger/
i386 job: https://doi-janky.infosiftr.net/job/multiarch/job/i386/job/rust/
amd64 job: https://doi-janky.infosiftr.net/job/multiarch/job/amd64/job/rust/

**Comment by waghanza** [comment]
Not sure it is normal @tristan-morris 

https://doi-janky.infosiftr.net/job/multiarch/job/amd64/job/_trigger/44699/ has started since 2 days 

Maybe someone at https://github.com/docker-library/oi-janky-groovy could help (@tianon or @yosifkit maybe)

**Comment by frederikhors** [comment]
I opened https://github.com/rust-lang/docker-rust/issues/148 and I think it is related.

I also noticed that each 1.70 build is MUCH slower than the 1.69 builds.

Maybe the same problem that makes these builds slow?

**Comment by tristan-morris** [comment]
1.70 is built and available now on Docker hub. 

Looks like Jenkins is much happier now. https://doi-janky.infosiftr.net/job/multiarch/job/amd64/job/tomee/ was disabled and the trigger job seems to be happily up-to-date.

**Comment by t-matsudate** [comment]
@tristan-morris Thank you for telling it to us! I checked that Docker's image ran on GitHub Actions. :)

**Comment by frederikhors** [comment]
@tristan-morris can you please explain this to noobs like me? What happened? What is going on right now?

**Comment by frederikhors** [comment]
@sfackler can you please explain what happened?

**Comment by sfackler** [comment]
The Docker-owned infrastructure that builds the images and publishes them to the Hub was backed up.

**Comment by frederikhors** [comment]
> The Docker-owned infrastructure that builds the images and publishes them to the Hub was backed up.

Ok, but the 1.70 was published AND it was a broken image. Is this the normal behavior?

**Comment by sfackler** [comment]
What does "broken" mean in this context?

**Comment by frederikhors** [comment]
Sorry, with broken I mean issues like:

- https://github.com/rust-lang/docker-rust/issues/148
- https://github.com/rust-lang/docker-rust/issues/149

and so on...

Why this happened?

**Comment by sfackler** [comment]
#148 someone already replied why:

> This was the problem I had yesterday. My binaries were built against i386 and then running in amd64, which led me to https://github.com/rust-lang/docker-rust/issues/147. The OP got in there a few minutes before :)

#149 includes literally zero information beyond "it fails", so I have no idea.

**Comment by frederikhors** [comment]
> My binaries were built against i386

because the image was broken. It was not building amd64 binaries. This is the question: why?

**Comment by frederikhors** [comment]
![image](https://github.com/rust-lang/docker-rust/assets/41120635/84afaf6f-89d8-4b40-a636-9e98331ba7d3)

**Comment by sfackler** [comment]
https://docs.docker.com/build/building/multi-platform/

**Comment by sfackler** [comment]
Can you rephrase that question with respect to the information included in the link I just posted?

**Comment by tianon** [comment]
If you want "amd64 _only_" (not "amd64 with i386 fallback"), you'll want/need to pin either to the content digest of the amd64 image or use `amd64/rust:xxx` explicitly (neither Docker nor containerd whose `--platform` syntax Docker supports/inherits support a "this exact platform" specification).

As noted in this thread, the underlying issue was that another build gummed up the official images build queue, and that's since been corrected.
